clippy_lints/src/methods/mod.rs

```diff
@@ -756,6 +757,32 @@ declare_clippy_lint! {
     "using `Iterator::step_by(0)`, which will panic at runtime"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for the use of `iter.nth(0)`.
+    ///
+    /// **Why is this bad?** `iter.nth(0)` is unnecessary, and `iter.next()`
+    /// is more readable.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust
```

Comment:
```suggestion
    /// ```rust
    /// # use std::collections::HashSet;
```

Reply:
yep just about to push it up, thanks for bearing with me!